tests/Operation/WatchFunctionalTest.php

```diff
@@ -588,6 +588,46 @@ public function testResumeTokenNotFoundAdvancesKey()
         $this->assertSame(2, $changeStream->key());
     }
 
+    public function testSessionPersistsAfterResume()
+    {
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $this->defaultOptions);
+
+        $changeStream = null;
+        $originalSession = null;
+        $sessionAfterResume = null;
+
+        // We want to ensure that the lsid of the initial aggregate matches the
+        // lsid of any aggregates after the change stream resumes. After
+        // PHPC-1152 is complete, we will ensure that the lsid of the initial
+        // aggregate matches the lsid of any subsequent aggregates and getMores.
+        (new CommandObserver)->observe(
+            function() use ($operation, &$changeStream) {
+                $changeStream = $operation->execute($this->getPrimaryServer());
+            },
+            function($changeStream) use (&$originalSession) {
+                if (isset($changeStream->aggregate)) {
+                    $originalSession = $changeStream->lsid;
```

Comment:
AFAIK `lsid` may include other fields, so you may want to only capture `lsid->id` (and/or possibly cast it to a hex string as I'm doing in the PHPC tests:

```
bin2hex((string) $command->lsid->id);
```

This makes output easier to digest if comparisons fail, since you're no longer dealing with raw binary strings and `MongoDB\BSON\Binary` objects.